py3status/modules/gazelle.py

```diff
@@ -0,0 +1,164 @@
+    format: See placeholders below (default 'Ratio: {ratio}')
+    password: Account password (default '')
+    thresholds: specify color thresholds to use
+        (default [(0, 'bad'), ('requiredratio', 'degraded'), (1, 'good')])
+    urgent_new_messages: set output urgent mode when there are new messages
+        (default False)
+    urgent_new_notifs: set output urgent mode when there are new notifications
+        (default False)
+    username: Account username (default '')
+
+Format placeholders:
+    {ratio} account ratio
+    {download} account download amount
+    {upload} account upload amount
+    {messages} number of unread messages
+    {notifs} number of new notifications
+
+@author raspbeguy <raspbeguy@hashtagueule.fr>
+@license MIT
+"""
+
+from py3status.exceptions import RequestException
+
+import re
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    baseurl = None
+    cache_timeout = 60
+    format = 'Ratio: {ratio}'
+    password = ''
+    thresholds = [(0, 'bad'), ('requiredratio', 'degraded'), (1, 'good')]
+    urgent_new_messages = False
+    urgent_new_notifs = False
+    username = ''
+
+    def _readable_size(self, size):
+        num = size
+        for unit in ['B', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB']:
+            if abs(num) < 1024.0:
+                return "%3.1f %s" % (num, unit)
+            num /= 1024.0
+        return "%.1f%s" % (num, 'YiB')
+
+    def _login(self):
+        self.py3.request(
+                self.baseurl+"/login.php",
+                data={'username': self.username, 'password': self.password},
+                headers={'User-Agent': 'Mozilla/5.0'},
+                cookiejar=self._cookiejar
+                )
+        try:
+            self._index()  # just to try if login info are good
+        except RequestException:
+            self.py3.log(
+                    "Cannot login to %s" % self.baseurl,
+                    level=self.py3.LOG_ERROR
+                    )
+            raise
+        self._cookiejar.save(self.baseurl)
+        self.py3.log(
+                "Saved cookie in storage",
+                level=self.py3.LOG_INFO
+                )
+
+    def _index(self):
+        resp = self.py3.request(
+                self.baseurl+"/ajax.php",
+                params={'action': "index"},
+                headers={
+                    'User-Agent': 'Mozilla/5.0'
+                    },
+                cookiejar=self._cookiejar
+                )
+        return resp.json()
+
+    def post_config_hook(self):
+        # Check if given URL is really an URL
+        if not isinstance(self.baseurl, str) or not re.search("^https?://", self.baseurl):
+            self.py3.error("Invalid URL")
+        self.baseurl = self.baseurl.strip('/')
+        if not isinstance(self.username, str) or not isinstance(self.password, str):
+            self.py3.error("Invalid login info")
```

Comment:
Good checking. The usual behavior / consistency for py3status modules is to start them off with `None` then check to make sure they're not `None`.

Reply:
Well, people might enter a list, a dict, a number... which would be incorrect.